### PR TITLE
Update release info for upcoming 3.4a2

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

The master branch is tracking 3.4.x. Update for upcoming alpha
release.

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [x] Passes warning tests
* [x] Passes all tests
* [x] Add comments such that someone without knowledge of the code could understand
* [x] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

**Comment by raffenet** [comment]
test:jenkins/ch4/most

File: RELEASE_NOTES [hunk 1]
```diff
@@ -6,9 +6,6 @@
 
  * Build Issues
    - CH4 will not build with Solaris compilers.
```
Comment (hzhou) [inline RELEASE_NOTES]: The master currently does build with Solaris : https://jenkins-pmrs.cels.anl.gov/job/mpich-master-ch4-ofi/

It only build with external libfabric and external ucx though.

File: RELEASE_NOTES [hunk 2]
```diff
@@ -6,9 +6,6 @@
...
- * Stability with large buffers
-   - CH4 may suffer from stability issues when operating on large
-     communication buffers (>= 2GB)
 
  * Dynamic process support
    - Requires fi_tagged capability in OFI netmod
```
Comment (hzhou) [inline RELEASE_NOTES]: Dynamic process does not work with ucx, right?

Comment (raffenet) [reply RELEASE_NOTES]: Yes, I should add a note about that.

File: RELEASE_NOTES (hunk 1 above)
Comment (raffenet) [reply RELEASE_NOTES]: OK, great. I'll not that and the limitations.

**Comment by raffenet** [comment]
test:jenkins/ch4/most

**Comment by raffenet** [comment]
I updated the Solaris bullet, added a note about 32-bit not working yet, and softened the language about configure without selecting a netmod.

**Pull request merged**